### PR TITLE
In 98 gestion des bonus

### PR DESCRIPTION
**Pull request by antoinepnt**
Red bonus is for blastPower, Green for speed and Black for nbBomb

### CODE REVIEW

File: src/components/Component.hpp [hunk 1]
```diff
@@ -43,7 +42,7 @@ namespace indie
         }
 
     protected:
-        Type _type = Type::TYPE_NB;
+        Type _type;
```
Comment (MauxC) [inline src/components/Component.hpp:45]: Why remove the default value of the Component type ?

File: src/components/IComponent.hpp [hunk 2]
```diff
@@ -27,20 +27,21 @@ namespace indie
...
-            PLAYER,
             ANIMATION,
             BOMB,
             TIMER,
             DESTRUCTIBLE,
             AI,
             TYPE_NB
         };
-        virtual Type getType() const = 0;
+        virtual const Type &getType() const = 0;
```
Comment (MauxC) [inline src/components/IComponent.hpp:44]: Why add reference of an enum ?

File: src/systems/GameSystem.cpp [hunk 3]
```diff
@@ -27,6 +29,10 @@
 #include "Scene.hpp"
 #include "Sprite.hpp"
 #include "String.hpp"
+#include "Bonus.hpp"
+#include "Model3D.hpp"
+#include "Sphere.hpp"
+#include "Grid.hpp"
```
Comment (MauxC) [inline src/systems/GameSystem.cpp:35]: Model3D and Grid are already imported

**Review by MauxC — CHANGES_REQUESTED** [review]

File: src/components/IComponent.hpp (hunk 2 above)
Comment (iDommel) [reply src/components/IComponent.hpp:44]: To avoid a copy, but maybe it's overkill

Comment (antoinepnt) [reply src/components/IComponent.hpp:44]: why not? this value is not gonna be modified, so better use it as const

Comment (MauxC) [reply src/components/IComponent.hpp:44]: Type is an enum so a number a reference is fatter

**Review by iDommel — APPROVED** [review]

**Review by MauxC — APPROVED** [review]

**Pull request merged**